marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -302,7 +298,22 @@ class Flow {
     }
 
     this.writeProvenanceData(jobId, flowName, stepDefinition, flowStep);
-    this.updateBatchDocument(flowName, flowStep, combinedOptions, items, writeTransactionInfo);
+
+    //write batch data if required. This is defered until after the batch is complete to handle onFailure properly
+    if (!(
+        combinedOptions.batchOutput === 'never' ||
+        (combinedOptions.batchOutput === 'onFailure' && !this.globalContext.failedItems.length) ||
+        combinedOptions.disableJobOutput
+      )) {
+
+      let batchStatus = "finished";
+      if (this.globalContext.failedItems.length) {
+        batchStatus = this.globalContext.completedItems.length ? "finished_with_errors" : "failed";
+      }
+
+      let batchDoc = this.datahub.jobs.insertBatch(jobDoc, flowStep, stepNumber, batchStatus, this.globalContext.completedItems, this.globalContext.batchErrors[0], combinedOptions);
+      this.globalContext.batchId = batchDoc.batch.batchId;
```

Comment:
I don't believe this is needed anymore - it was used so that updateBatch knew what the batchId was.